pulpcore/cli/ansible/distribution.py

```diff
@@ -0,0 +1,113 @@
+from typing import Optional, Dict, Any
+
+import click
+
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleDistributionContext,
+    PulpAnsibleRepositoryContext,
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "distribution_type",
+    type=click.Choice(["ansible"], case_sensitive=False),
+    default="ansible",
+)
+@pass_pulp_context
+@click.pass_context
+def distribution(ctx: click.Context, pulp_ctx: PulpContext, distribution_type: str) -> None:
+    if distribution_type == "ansible":
+        ctx.obj = PulpAnsibleDistributionContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+distribution.add_command(list_entities)
+distribution.add_command(show_by_name)
+distribution.add_command(destroy_by_name)
+
+
+# TODO Add content_guard option
+@distribution.command()
+@click.option("--name", required=True)
+@click.option("--base-path", required=True)
+@click.option("--repository", help="name of repository")
+@click.option("--version", help="version of repository, leave blank for always latest")
+@pass_entity_context
+@pass_pulp_context
+def create(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    name: str,
+    base_path: str,
+    repository: Optional[str],
+    version: Optional[str],
+) -> None:
+    body: Dict[str, Any] = {"name": name, "base_path": base_path}
+    repo: Dict[str, Any] = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
+    if version:
+        body["repository_version"] = f'{repo["versions_href"]}{version}/'
+    elif repository:
+        body["repository"] = repo["pulp_href"]
+    result = distribution_ctx.create(body=body)
+    distribution = distribution_ctx.show(result["created_resources"][0])
+    pulp_ctx.output_result(distribution)
+
+
+@distribution.command()
+@click.option("--name", required=True)
+@click.option("--base-path")
+@click.option("--repository", type=str, default=None)
+@click.option("--version", type=str, default=None)
+@pass_entity_context
+@pass_pulp_context
+def update(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    name: str,
+    base_path: Optional[str],
+    repository: Optional[str],
+    version: Optional[str],
+) -> None:
+    """
+    To remove repository or repository_version set --repository to ""
+    """
+    dist_body = distribution_ctx.find(name=name)
+    href = dist_body["pulp_href"]
+    body = dict()
+    if base_path:
+        body["base_path"] = base_path
+    if repository is not None:
```

Comment:
I think this command would benefit from a comment since I think you are misunderstanding how it works. The two options '--repository' (repository's name) and '--version' (a repository's version number) are not mutually exclusive. If you only specify repository then it'll set the repository field. If you specify repository and version, it'll set the repository_version field.  This is also how it works in the create command.

Reply:
Oh right, version is just the number.
Maybe we should error on a version number without repository then.